packages/wrangler/src/triggers/deploy.ts

```diff
@@ -90,24 +94,35 @@ export default async function triggersDeploy(
 	const uploadMs = Date.now() - start;
 	const deployments: Promise<string[]>[] = [];
 
+	const deploymentInSync = deployToWorkersDev === available_on_subdomain;
+	const previewsInSync =
+		config.workers_dev_previews === previews_available_on_subdomain;
+
 	if (deployToWorkersDev) {
 		// Deploy to a subdomain of `workers.dev`
 		const userSubdomain = await getWorkersDevSubdomain(accountId);
-		const scriptURL =
+
+		const deploymentURL =
 			props.legacyEnv || !props.env
 				? `${scriptName}.${userSubdomain}.workers.dev`
 				: `${envName}.${scriptName}.${userSubdomain}.workers.dev`;
-		if (!available_on_subdomain) {
+
+		if (deploymentInSync && previewsInSync) {
+			deployments.push(Promise.resolve([deploymentURL]));
+		} else {
 			// Enable the `workers.dev` subdomain.
 			deployments.push(
 				fetchResult(`${workerUrl}/subdomain`, {
 					method: "POST",
-					body: JSON.stringify({ enabled: true }),
+					body: JSON.stringify({
+						enabled: true,
+						previews_enabled: config.workers_dev_previews,
+					}),
 					headers: {
 						"Content-Type": "application/json",
 					},
 				})
-					.then(() => [scriptURL])
+					.then(() => [deploymentURL])
 					// Add a delay when the subdomain is first created.
```

Comment:
Incidentally, we should probably remove this. Avoiding negative dns cache is important during the first ~30 seconds of *subdomain registration*, but is not relevant when enabling the subdomain route of a particular Worker.